### ISSUE TITLE
Implementing dynamic importance sampling as a bias

### ISSUE THREAD

**Issue by dotsdl** [plumed/plumed2#135]
I am trying to implement [dynamic importance sampling](http://dx.doi.org/10.1002/jcc.21564) as a biasing algorithm for plumed, and I'm using the code for [ABMD](http://plumed.github.io/doc-v2.1/user-doc/html/_a_b_m_d.html) as the starting point, since it's algorithmically a bit similar. I have a few questions, however:

1. Is it possible to set the forces calculated by the MD engine to zero from within PLUMED? Is there at least a function to obtain these forces as an array?

2. Is there a function for setting/randomizing the velocities of the particles in the system within PLUMED?

**Comment by GiovanniBussi** [comment]
1. Forces are not made available as of now. However, if you just need to set them to zero, just add a bias on the total energy. In the input file it would be
e: ENERGY
RESTRAINT ARG=e SLOPE=-1 KAPPA=0.0
If you want to do it inside the collective you can copy some lines from colvar/Energy.cpp
2. This has been discussed in the mailing list. It is not easy to do it in a way that is compatible with multiple MD codes. In case many people want this we could add the possibility to pass velocities (from the MD code) and to receive them (from a CV). However, this requires some work.

I hope this helps! Giovanni

**Comment by dotsdl** [comment]
Ah. This helps a ton! Thanks! I'm thinking of some alternative ways of implementing the basic algorithm given these constraints, and I think I have at least one solution!

**Comment by dotsdl** [comment]
I've built a ``Bias`` that roughly implements dynamic importance sampling. However, the code makes use of the [random](http://www.cplusplus.com/reference/random/) library from the standard lib, and on compile I get:

```
In file included from /usr/include/c++/4.9.2/random:35:0,
                 from DIMS.cpp:27:
/usr/include/c++/4.9.2/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support for the \
  ^
```

I'm interpreting this as the compiler complaining that using ``random`` requires C++11. What minimum version of the C++ standard is PLUMED2 intended to support? I couldn't find this information in the [developer docs](http://plumed.github.io/doc-v2.1/developer-doc/html/index.html).

The current state of the implementation can be viewed [here](https://github.com/dotsdl/plumed2/blob/bias-dims/src/bias/DIMS.cpp).

**Comment by GiovanniBussi** [comment]
Perhaps your compiler does not support C++11 (or you have to add some flag to CPPFLAGS for it to support C++11).

If it's for your own usage feel free to use C++11. If you plan to contribute back the implementation for us to include it in the main distribution, it would be better to stick to C++03 (for maximum portability) or, alternatively, protect your class with some cpp directive so that it can easily be excluded when C++11 is not supported.
